gameSource/hetuwmod.cpp

```diff
@@ -1730,11 +1730,60 @@ void HetuwMod::SayStep() {
 void HetuwMod::Say(const char *text) {
 	if (bTeachLanguage) bTeachLanguage = false;
 
-	char *msg = new char[strlen(text)+1];
-	strcpy(msg, text);
+	char *msg = new char[strlen(text)*2+1];
+	encodeDigits(text, msg);
 	sayBuffer.push_back(msg);
 }
 
+// Encode digits using 0 = ?A, 1 = ?B, 2 = ?C, etc. due to server restrictions
+void HetuwMod::encodeDigits(const char *plain, char *encoded) {
+	bool questionMark = false;
+	int j = 0;
+	size_t len = strlen(plain);
+	for (size_t i=0; i<len; i++) {
+		if (isdigit(plain[i])) {
```

Comment:
`ctypes.h` is a minefield that's usually best avoided these days. `isdigit()` is actually the only function in there not affected by locale, but I had to look that up to confirm. If you mean a range of ASCII character values it's usually best to just do it explicitly to make life easier on future readers who will worry about C locale nonsense.